### PR TITLE
Add status to bigquery_job

### PR DESCRIPTION
**Pull request by kanterov**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/8357

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigquery: added `status` field to `google_bigquery_job`
```

### CODE REVIEW

File: mmv1/products/bigquery/api.yaml [hunk 1]
```diff
@@ -978,6 +978,40 @@ objects:
             description: |
               The geographic location of the job. The default value is US.
             default_value: 'US'
+      - !ruby/object:Api::Type::NestedObject
+        name: 'status'
+        output: true
```
Comment (kanterov) [inline mmv1/products/bigquery/api.yaml:983]: Not sure if this is needed. Because of that terraform generator doesn't set `maxItems: 1`

**Comment by kanterov** [comment]
@googlebot I signed it!

**Review by slevenick — COMMENTED** [review]
/gcbrun

Rerunning jobs as release note is fixed now

**Comment by slevenick** [comment]
/gcbrun

Apparently review comments don't trigger this

**Comment by kanterov** [comment]
I fixed my setup so I can run tests locally and now I can see what's wrong. Fixing it now and going to update the code soon.

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like this causes a panic during the bigquery tests:

```
panic: status.0.error_result: '' expected type 'string', got unconvertible type 'map[string]interface {}'
```

Looks like the `errorResult` is this structure: https://cloud.google.com/bigquery/docs/reference/rest/v2/ErrorProto

rather than just a string

**Comment by kanterov** [comment]
@slevenick sorry, totally confused it. Now tests pass locally, is it possible to re-trigger GCB?

**Comment by kanterov** [comment]
I can't reproduce the failure of `TestAccBigQueryJob_bigqueryJobLoadExample` locally, is there any way to see logs?

`TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample` and `TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityCmekExample` seem to be failing in main branch as well.

**Comment by slevenick** [comment]
> I can't reproduce the failure of `TestAccBigQueryJob_bigqueryJobLoadExample` locally, is there any way to see the logs?
> 
> `TestAccActiveDirectoryDomainTrust_activeDirectoryDomainTrustBasicExample` and `TestAccPrivatecaCertificateAuthority_privatecaCertificateAuthorityCmekExample` seem to be failing in the main branch as well.

Yeah, the other two tests are unrelated to your changes, ignore them.

I'm still seeing this failure for the bigquery job:
```
TestAccBigQueryJob_bigqueryJobLoadExample: provider_test.go:270: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
(map[string]string) (len=1) {
(string) (len=14) "status.0.state": (string) (len=4) "DONE"
}
(map[string]string) (len=1) {
(string) (len=14) "status.0.state": (string) (len=7) "RUNNING"
}
```

it looks like the job status is changing between creating the job & the import step running. This may be flaky due to the time it takes the job to finish. We probably want to ignore the status when running the import step due to this, as it should never be set by the user. 

You can add `status.0.state` to the `ImportStateVerifyIgnore` for the bigquery job tests and it should fix this

**Comment by kanterov** [comment]
@slevenick thanks for the clarifications, I've updated tests, including auto-generated ones. Should we /gcbrun?

**Review by slevenick — APPROVED** [review]
Looks good to me! Thanks

**Comment by kanterov** [comment]
@slevenick thanks a lot for your support, I learned a lot!

**Pull request merged**